### PR TITLE
updated deprecated makedocs call: format; clean build dir; aligned code

### PR DESCRIPTION
**Pull request by PaulXiCao**
- major change: updated makedoc call. `format = :html` is deprecated.
- minor
  - cleaning build dir by default is useful
  - aligned code 

PS: this is in reference to issue #159 .

### CODE REVIEW

File: docs/make.jl [hunk 1]
```diff
@@ -1,18 +1,16 @@
...
     ],
 )
 
 deploydocs(
-    repo = "github.com/JuliaMath/SpecialFunctions.jl.git",
-    target = "build",
-)
+    repo    = "github.com/JuliaMath/SpecialFunctions.jl.git",
+    target  = "build",
+)
```
Comment (giordano) [inline docs/make.jl]: Please, leave the newline at the end of the line.

Comment (PaulXiCao) [reply docs/make.jl]: Ok, just updated my editor's settings.

Comment (PaulXiCao) [reply docs/make.jl]: appended newline to file in new commit

**Comment by simonbyrne** [comment]
Thanks @PaulXiCao. A few points:

1) I think this information would be better incorporated into the docstrings. These could then be grouped into headings as you've done here. 
2) Whether there is a BigFloat implementation is really an implementation detail, i.e. you can write something like:
```
## Implementation details

- `Float32`/`Float64`: C standard math library (libm)
- `BigFloat`: [MPFR](https://www.mpfr.org/)
```

**Comment by PaulXiCao** [comment]
@simonbyrne I cannot find a good workflow. The documenter.jl does not use my uncommited docstrings. Do you know any way?

**Comment by simonbyrne** [comment]
Change to the docs directory, start julia with `julia --project`, then do `]dev ..`

**Comment by PaulXiCao** [comment]
@simonbyrne could you give a short feedback on my commits?
I have done the following for all functions in erf.jl and ellip.jl

- added documentation
- cleaned the source code 
- added tests for every possible case (data types and each piecewise definition)

My questions

1. Any feedback? Would you like something done differently?
2. If I do the same for other files would you pull it at some point? 
3. Should I split my work into different merge requests or just keep pushing to this one?

**Comment by PaulXiCao** [comment]
Closed by mistake... *facepalm*

File: docs/src/functions_overview.md [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+## Elliptic Integrals
+
+| Function | Description |
+|:-------- |:----------- |
+| [`ellipk(m)`](@ref SpecialFunctions.ellipk)    | [complete elliptic integral of 1st kind](https://en.wikipedia.org/wiki/Elliptic_integral#Notational_variants) ``K(m)``  |
+| [`ellipe(m)`](@ref SpecialFunctions.ellipe)    | [complete elliptic integral of 2nd kind](https://en.wikipedia.org/wiki/Elliptic_integral#Complete_elliptic_integral_of_the_second_kind) ``E(m)`` |
+
+
+# WIP not grouped
+| Function                                                      | Description                                                                                                                                                     |
```
Comment (simonbyrne) [inline docs/src/functions_overview.md]: I would suggest the following groupings:
- Airy, Bessel and Hankel functions
- Gamma and related functions (anything involving `gamma`, `beta`, `logfactorial`)
- Trigonmetric integral (`sinint`, `cosint`)
- Riemann zeta and related functions (`zeta`, `eta`)

Comment (PaulXiCao) [reply docs/src/functions_overview.md]: Sounds good. I'm doing the trigonometric integrals right now. Will also group the other ones but their documentation will take more time..

File: docs/src/functions_overview.md [hunk 3]
```diff
@@ -62,6 +52,17 @@ Here the *Special Functions* are listed according to the structure of [NIST Digi
 | [`besselix(nu,z)`](@ref SpecialFunctions.besselix)            | scaled modified Bessel function of the first kind of order `nu` at `z`                                                                                          |
 | [`besselk(nu,z)`](@ref SpecialFunctions.besselk)              | modified [Bessel function](https://en.wikipedia.org/wiki/Bessel_function) of the second kind of order `nu` at `z`                                               |
 | [`besselkx(nu,z)`](@ref SpecialFunctions.besselkx)            | scaled modified Bessel function of the second kind of order `nu` at `z`                                                                                         |
+
+## Gamma and Related Functions
```
Comment (simonbyrne) [inline docs/src/functions_overview.md]: "Related" shouldn't be capitalized.

File: docs/src/functions_overview.md [hunk 4]
```diff
@@ -71,3 +72,9 @@ Here the *Special Functions* are listed according to the structure of [NIST Digi
 | [`logbeta(x,y)`](@ref SpecialFunctions.logbeta)                                          | accurate `log(beta(x,y))` for large `x` or `y`      |
 | [`logabsbeta(x,y)`](@ref SpecialFunctions.logabsbeta)                                          | accurate `log(abs(beta(x,y)))` for large `x` or `y`     |
 | [`logabsbinomial(x,y)`](@ref SpecialFunctions.logabsbinomial)                                          | accurate `log(abs(beta(x,y)))` for large `x` or `y`     |
+
+## Riemann Zeta and Related Functions
```
Comment (simonbyrne) [inline docs/src/functions_overview.md]: Same: "Related Functions"  shouldn't be capitalized.

Comment (PaulXiCao) [reply docs/src/functions_overview.md]: fixed

**Comment by simonbyrne** [comment]
Looks much better, thanks!

**Pull request merged**